pkg/openstack/loadbalancer.go

```diff
@@ -451,6 +451,57 @@ func getListenerProtocol(protocol corev1.Protocol, svcConf *serviceConfig) liste
 	}
 }
 
+func (lbaas *LbaasV2) createUnpopulatedLoadBalancer(name, clusterName string, service *corev1.Service, svcConf *serviceConfig) (*loadbalancers.LoadBalancer, error) {
+	createOpts := loadbalancers.CreateOpts{
+		Name:        name,
+		Description: fmt.Sprintf("Kubernetes external service %s/%s from cluster %s", service.Namespace, service.Name, clusterName),
+		Provider:    lbaas.opts.LBProvider,
+	}
+
+	if svcConf.supportLBTags {
+		createOpts.Tags = []string{svcConf.lbName}
+	}
+
+	if svcConf.flavorID != "" {
+		createOpts.FlavorID = svcConf.flavorID
+	}
+
+	if svcConf.availabilityZone != "" {
+		createOpts.AvailabilityZone = svcConf.availabilityZone
+	}
+
+	vipPort := getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerPortID, "")
+	lbClass := lbaas.opts.LBClasses[svcConf.configClassName]
+	if vipPort != "" {
+		createOpts.VipPortID = vipPort
+	} else {
+		if lbClass != nil && lbClass.SubnetID != "" {
+			createOpts.VipSubnetID = lbClass.SubnetID
+		} else {
+			createOpts.VipSubnetID = svcConf.lbSubnetID
+		}
+		if lbClass != nil && lbClass.NetworkID != "" {
+			createOpts.VipNetworkID = lbClass.NetworkID
+		} else if svcConf.lbNetworkID != "" {
+			createOpts.VipNetworkID = svcConf.lbNetworkID
+		} else {
+			klog.V(4).Infof("network-id parameter not passed, it will be inferred from subnet-id")
+		}
+	}
+	// For external load balancer, the LoadBalancerIP is a public IP address.
+	loadBalancerIP := service.Spec.LoadBalancerIP
+	if loadBalancerIP != "" && svcConf.internal {
+		createOpts.VipAddress = loadBalancerIP
+	}
```

Comment:
Looks like this whole fragment is duplicated in lines 506-549. Could you extract it to another function and use it in both places? Something like `getOctaviaLoadBalancerCreateOpts()`.

Reply:
I ended up switching this around a bit to reuse the original function and just add the "fully-populated" createOpts only when the provider doesn't require the serial-style requests.

Happy to do as you suggest here if this approach isn't as good in your opinion.

Reply:
Oh no, this is nice and clean, great idea!